crates/sui-json-rpc/src/routing_layer.rs

```diff
@@ -251,3 +250,15 @@ mod response {
             .expect("Unable to parse response body for type conversion")
     }
 }
+
+pub fn is_json(request: &hyper::Request<hyper::Body>) -> bool {
+    request
+        .headers()
+        .get(http::header::CONTENT_TYPE)
+        .and_then(|val| val.to_str().ok())
+        .map_or(false, |content| {
+            content.eq_ignore_ascii_case("application/json")
+                || content.eq_ignore_ascii_case("application/json; charset=utf-8")
+                || content.eq_ignore_ascii_case("application/json;charset=utf-8")
```

Comment:
nit: shall we just convert content to lower case and check if it contains `application/json`?

Reply:
this is "borrowed" from jsonrpsee codebase, I will leave it as it for now to make the behaviour consistent.